File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/keyfederation/BatchTagService.java
package uk.nhs.nhsx.keyfederation;

import uk.nhs.nhsx.keyfederation.upload.lookup.UploadKeysResult;

import java.time.LocalDate;
import java.util.Optional;

public interface BatchTagService {

    Optional<UploadKeysResult> getLastUploadState();

    void updateLastUploadState(Long lastUploadTimestamp);

    Optional<FederationBatch> getLatestFederationBatch();

    void updateLatestFederationBatch(FederationBatch batch);

    class FederationBatch {
        public final BatchTag batchTag;
        public final LocalDate batchDate;

        public FederationBatch(BatchTag batchTag, LocalDate batchDate) {
            this.batchTag = batchTag;
            this.batchDate = batchDate;
        }
    }
}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/keyfederation/BatchTagService.kt
package uk.nhs.nhsx.keyfederation

import uk.nhs.nhsx.keyfederation.upload.lookup.UploadKeysResult
import java.time.Instant
import java.util.*

interface BatchTagService {
    fun lastUploadState(): Optional<UploadKeysResult>
    fun updateLastUploadState(lastUploadTimestamp: Instant)
    fun latestFederationBatch(): Optional<FederationBatch>
    fun updateLatestFederationBatch(batch: FederationBatch)
}
